packages/cli/src/commands/init.ts

```diff
@@ -475,282 +405,406 @@ async function processInitForm(
-  let contractNameFromEtherscan: string | undefined = undefined;
-
   try {
-    const { protocol } = await prompt.ask<{ protocol: ProtocolName }>({
-      type: 'select',
-      name: 'protocol',
-      message: 'Protocol',
-      choices: protocolChoices,
-      skip: protocolChoices.includes(String(initProtocol) as ProtocolName),
-      result: value => {
-        if (initProtocol) {
-          initDebugger.extend('processInitForm')('initProtocol: %O', initProtocol);
-          return initProtocol;
-        }
-        initDebugger.extend('processInitForm')('protocol: %O', value);
-        return value;
-      },
+    const registry = await loadRegistry();
+    const contractService = new ContractService(registry);
+
+    const networks = sortWithPriority(
+      registry.networks,
+      n => n.issuanceRewards,
+      (a, b) => registry.networks.indexOf(a) - registry.networks.indexOf(b),
+    );
+
+    const networkToChoice = (n: Network) => ({
+      name: n.id,
+      value: `${n.id}:${n.shortName}:${n.fullName}`.toLowerCase(),
+      hint: `· ${n.id}`,
+      message: n.fullName,
     });
 
-    const protocolInstance = new Protocol(protocol);
-    const isComposedSubgraph = protocolInstance.isComposedSubgraph();
-    const isSubstreams = protocol === 'substreams';
-    initDebugger.extend('processInitForm')('isSubstreams: %O', isSubstreams);
+    const formatChoices = (choices: ReturnType<typeof networkToChoice>[]) => {
+      const shown = choices.slice(0, 20);
+      const remaining = networks.length - shown.length;
+      if (remaining == 0) return shown;
+      if (shown.length === choices.length) {
+        shown.push({
+          name: '--other--',
+          value: '',
+          hint: '· other network not on the list',
+          message: `Other`,
+        });
+      }
+      return [
+        ...shown,
+        {
+          name: ``,
+          disabled: true,
+          hint: '',
+          message: `< ${remaining} more networks >`,
+        },
+      ];
+    };
 
-    const { subgraphName } = await prompt.ask<{ subgraphName: string }>([
-      {
-        type: 'input',
-        name: 'subgraphName',
-        message: 'Subgraph slug',
-        initial: initSubgraphName,
+    let network = networks[0];
+    let protocolInstance: Protocol = new Protocol('ethereum');
+    let isComposedSubgraph = false;
+    let isSubstreams = false;
+    let subgraphName = initSubgraphName ?? '';
+    let directory = initDirectory;
+    let ipfsNode: string = '';
+    let source = initContract;
+    let contractName = initContractName;
+    let abiFromFile: EthereumABI | undefined = undefined;
+    let abiFromApi: EthereumABI | undefined = undefined;
+    let startBlock: string | undefined = undefined;
+    let spkgPath: string | undefined;
+    let spkgCleanup: (() => void) | undefined;
+    let indexEvents = initIndexEvents;
+
+    const promptManager = new PromptManager();
+
+    promptManager.addStep({
```

Comment:
Since users will be coming of the old `init` behavior of having a huge list of networks to select from that they navigate "up/down", we need to teach them the new UX.

Recommending adding more details on how to use the prompt such as `Type to search for network` or similar.

Reply:
Good point. Added hint on the bottom:
`< XX more - type to filter >`